examples/list_files.rb

```diff
@@ -1,17 +1,15 @@
 require 'gphoto2'
 
-# List all files in tree form.
-
-def visit(folder, level = 0)
-  indent = '  ' * level
+# List all files
 
+def visit(folder)
   files = folder.files
-  puts "#{indent}#{folder.root? ? "/ (root)" : folder.path} (#{files.size} files)"
+  puts "#{folder.root? ? "/ (root)" : folder.path} (#{files.size} files)"
 
-  folder.folders.each { |child| visit(child, level + 1) }
-  indent << '  '
+  folder.folders.each { |child| visit(child) }
```

Comment:
This traversal should be after completely visiting the node, else you end up printing a series of empty lines coming up from the stack.